pkg/api/types.go

```diff
@@ -505,14 +505,15 @@ func (l *VolumeList) Set(value string) error {
 		return errors.New("invalid format, must be source:destination")
 	}
 	var mount []string
-	pos := strings.LastIndex(value, ":")
+	pos := strings.IndexAny(value, ":")
 	if pos == -1 {
 		mount = []string{value, ""}
 	} else {
 		mount = []string{value[:pos], value[pos+1:]}
 	}
 	mount[0] = strings.Trim(mount[0], `"'`)
 	mount[1] = strings.Trim(mount[1], `"'`)
+
 	s := VolumeSpec{Source: filepath.Clean(mount[0]), Destination: filepath.ToSlash(filepath.Clean(mount[1]))}
 	if IsInvalidFilename(s.Source) || IsInvalidFilename(s.Destination) {
```

Comment:
No, since we don't check for ':', both source and destination would pass this check. However, if docker doesn't like either filename, it will fail. For example:
```
 $ s2i build https://github.com/pmorie/simple-ruby openshift/ruby-20-centos7 test-ruby-app --volume="/tmp/asdf:z:/home/bundle"
Your branch is up-to-date with 'origin/master'.
ERROR: An error occurred: Error response from daemon: Invalid bind mount spec "/tmp/asdf:z:/home/bundle": volumeinvalidmode: invalid mode: "/home/bundle"
```
I'll expand the PR description to be more clear